src/sknnr_spatial/datasets/_base.py

```diff
@@ -62,6 +62,22 @@ def _load_rasters_to_array(file_paths: list[Path]) -> NDArray:
     return arr
 
 
+@overload
+def load_swo_ecoplot(
+    as_dataset: Literal[True] = True,
+    large_rasters: bool = False,
+    chunks: Any = None,
+) -> tuple[xr.Dataset, pd.DataFrame, pd.DataFrame]: ...
+
+
+@overload
+def load_swo_ecoplot(
+    as_dataset: Literal[False] = False,
+    large_rasters: bool = False,
+    chunks: Any = None,
+) -> tuple[NDArray, pd.DataFrame, pd.DataFrame]: ...
```

Comment:
Super cool!  Strange to see three different "definitions" of the same function, but really nice to have clear typing information.